planning a trip to a cafe #73

sanioooook opened the pull request:
resolve #73

sanioooook committed:
add pages for adding places and displaying them
changed the food path to eating
add default image if the image didn't come from Google API

sanioooook committed:
Merge branch 'dev' of https://github.com/horba/my-trip into feature/73

sanioooook committed:
add a cafe planning component, add a Stor file,
add a controller, service, repository,
and scheduledPlaceToEat model to the server.
removed some warnings during build (extra using directives) on server
fixed an inactive swagger
(one controller did not explicitly specify
that the get method is get and not for example post)
it compiles and run,
but it doesn't work as it should, and file loading doesn't work

sanioooook committed:
fixed after loading the avatar the path
did not get into the db
making my own method for uploading multiple files at the same time

sanioooook committed:
added bad editing
scheduled displays don't work

sanioooook committed:
resove #73
added: filter for Google places types
interfaces for repositories and services (irepositorycontext)
added constants for scheduledPlaceToEat files

sanioooook committed:
Merge branch 'dev' of https://github.com/horba/my-trip into feature/73

sanioooook committed:
fix reloading the page after logout of the account

horba commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+          }
+        });
+    }
+  }
```
provide in api service method UploadFile() and add headers there

horba commented on front/src/views/user/eating/eating.view.html:
```diff
@@ -0,0 +1,109 @@
+          max-height="150" max-width="259" />
+        <v-img v-else :aspect-ratio="16/9" 
+          :lazy-src="defaultImage"
+          src="../../../assets/default-image.jpg"
```
@/assets

sanioooook replied:
ok

horba commented on front/src/filters/google-place-types-filter.js:
```diff
@@ -0,0 +1,27 @@
+    } else {
+      return type;
+    }
+  }
```
you can just add to root of translations keys:
```
"eating.meal_takeaway": "на виніс",
"meal_delivery": "доставка їжі"
```

and then in template
`<tag>{{$t(tplace.googleDeatails.type)</tag>`

sanioooook replied:
ok

horba commented on front/src/store/modules/auth.store.js:
```diff
@@ -1,4 +1,5 @@
 import api from '@api';
+import router from './../../router/index.js';
```
don't we have `@router`?

sanioooook replied:
ok

horba commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+            scheduledPlaceToEat.fileNames.forEach(fileName => {
+              scheduledPlaceToEat.files
+                .push(`${baseUrl}/${SERVER_SCHEDULED_PLACE_TO_EAT_PATH}/${fileName}`);
+            });
```
it should be in mapping dto logic - not at front

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+            return valid ? true : 'File size should be less than 10 MB!';
+          } else {
+            return true;
+          }
```
could it be useful for other components? if yes - it is nice to have mixin

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+            return v.length <= 10 || 'Количество вложений не должно быть больше 10';
+          } else {
+            return true;
+          }
```
the same

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+
+      const [month, day, year] = date.split('/');
+      return `${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`;
+    },
```
why don\t use date-fns formatting? might, you could use filter also

sanioooook replied:
as far as I understand date-fns translates time from numbers to a string, but I don't need it (I don't need 01.07.2020 to become 1 July 2020), I then can't pass it to the database, I need to not just display it for review, but also pass it later

horba replied:
it is not only about string view it is about date formats which exists as string and digits
please, read about date-time formats (it is common standart - not imagination of date-fns)
you could just write "yyyy-mm-dd"

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+            this.$store.dispatch('eating/uploadNewFilesEating', {
+              files: this.files,
+              eatingId: payload.id
+            });
```
```
this.$store.dispatch('eating/uploadNewFilesEating', { files, eatingId: payload.id }
```

sanioooook replied:
![image](https://user-images.githubusercontent.com/33792718/86187067-9aa61500-bb43-11ea-8125-a3e14ae4d7a0.png)

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+        googlePlaceId: this.googlePlaceId,
+        lat: this.lat,
+        lng: this.lng
+      };
```
```
const payload = { dateTime: `${this.date}T${this.time}`, namePlace, notes, link, googlePlaceId, lat, lng }
```

sanioooook replied:
if I'm remove 'this', the value will not be found (it will not be compiled), if I'm remove the property name and 'this.' - the same mistake will be made

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+              files: this.files,
+              eatingId: payload.id
+            });
+            setTimeout(() => this.$router.push({ name: 'ScheduleEatingPlace' }), 3000);
```
why do you need timeout? might you need nextTick()

demosglok replied:
i think we can just use .then

this.$store.dispatch(...)
.then(() => {
...
return this.$store.dispatch(...)
})
.then(() => this.$router.push(...))
.catch(e => handleException(e));

sanioooook replied:
snackBar doesn't have time to show up

horba replied:
it was absolutely not obvious 
the right thing is to make 
```
const SNACK_BAR_TIMEOUT = 4000;
// ...
showSnackbar(color, message) {
this.snackBarColor = color;
  this.snackBarMessage = message;
  this.showSnackBar = true;
  return new Promise(function(resolve, reject) {
    setTimeout(() => resolve(true), SNACK_BAR_TIMEOUT);
  });
}

//..

this.$store.dispatch(...)
  .then(() => {
    ...
    return this.$store.dispatch(...)
  })
  .then(showSnackBar)
  .then(() => this.$router.push(...))
  .catch(e => handleException(e));
```

but actually it is not pretty well to make people to wait page opening because it need to show message
this snackbar shouldn't belong to your view - it should be at common layout as container component (or clever component) and have actions and state in store - something like notification module

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+              files: this.files,
+              eatingId: r.data
+            });
+            setTimeout(() => this.$router.push({ name: 'ScheduleEatingPlace' }), 3000);
```
why do you need timeout? might you need nextTick()

demosglok replied:
just chained .then, no "setTimeout" here

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+          .catch(error => {
+            this.text = error;
+            this.snackbar = true;
+            this.color = 'error';
```
code dublication
might, you need
```
onSaveError(error) {
    this.text = error;
    this.snackbar = true;
    this.color = 'error';
}

/// ...
.catch(onSaveError)

horba replied:
the same with success
might, you need
```
onSaveSuccess(isEditMode) {
    this.text = isEditMode ? this.$t('eatingNew.successfullySaved') :  this.$t('eatingNew.successfullyEdited');;
    this.snackbar = true;
    this.color = 'success';
}

/// ...
then(response => {
  onSaveSuccess(isEditMode);
  return store.dispatch();
});
```

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+        lat: this.lat,
+        lng: this.lng
+      };
+      if (this.$route.params.id) {
```
computed property isEditMode

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+      files: new FormData(),
+      text: '',
+      snackbar: false,
+      color: ''
```
color of what? too common name

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+      lat: 0,
+      lng: 0,
+      files: new FormData(),
+      text: '',
```
what text? too common name

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+            this.color = 'error';
+          });
+      }
+    },
```
do you realy need different text message? and do you really need different server controllers?

sanioooook replied:
not understand this comment

horba commented on front/src/views/user/eating/eating.view.html:
```diff
@@ -0,0 +1,109 @@
+        :class="{ 'active-card': place.serverInfo.id === selectedId }">
+        <v-img v-if="place.googleDetails && place.googleDetails.photos" 
+          :aspect-ratio="16/9"
+          :lazy-src="defaultImage" :src="place.googleDetails.photos[0].getUrl(259)"
```
getUrl - fullPath filter (it seems we have one)

demosglok replied:
what a magic number 259. what does it mean?

sanioooook replied:
259 - size image

horba replied:
const - and name it well, please

horba commented on front/src/views/user/eating/eating.view.html:
```diff
@@ -0,0 +1,109 @@
+      </GmapMap>
+    </v-col>
+  </v-row>
+  <v-snackbar v-model="snackbar" :color="color" multi-line :timeout="3000" top>
```
snackback-color
snackbar-message

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating.view.js:
```diff
@@ -0,0 +1,127 @@
+          this.text = this.$t('eating.successfullyDelete');
+          this.snackbar = true;
+          this.color = 'success';
+          setTimeout(() => this.responseDataFromServer(), 3000);
```
it is strange, you need everywhere timeouts

demosglok replied:
its because he can't chain promises

remove setTimeout from here plz. use ".then"

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating.view.js:
```diff
@@ -0,0 +1,127 @@
+        isOpen = place.opening_hours.isOpen();
+      } catch (e) {
+        isOpen = true;
+      }
```
why do you need try catch

demosglok replied:
why "isOpen=true" for exception?

sanioooook replied:
because place.opening_hours can be undefined

ivhorodko replied:
`if (!place.opening_hours) {
          isOpen = place.opening_hours.isOpen();
}`

Or something like that

horba replied:
```
isOpen() {
  return place.opening_hours && place.opening_hours.isOpen();
}
```

horba commented on server/Entities/AttachmentFileEatingRepository.cs:
```diff
@@ -0,0 +1,50 @@
+{
+  public class AttachmentFileEatingRepository: IAttachmentFileEatingRepository
+  {
+    private readonly IRepositoryContext RepositoryContext;
```
_repositoryContext

horba replied:
for all private readonly fields

sanioooook replied:
ok

horba commented on server/Entities/ScheduledPlaceToEatRepository.cs:
```diff
@@ -0,0 +1,50 @@
+{
+  public class ScheduledPlaceToEatRepository: IScheduledPlaceToEatRepository
+  {
+    private readonly IRepositoryContext RepositoryContext;
```
_repositoryContext

horba commented on server/Entities/ScheduledPlaceToEatRepository.cs:
```diff
@@ -0,0 +1,50 @@
+    {
+      RepositoryContext.ScheduledPlacesToEat.Add(scheduledPlaceToEat);
+      RepositoryContext.SaveChanges();
+      return scheduledPlaceToEat.Id;
```
SaveChanged - mutable function? it is not pretty good

horba replied:
what if saving was unsuccess?

sanioooook replied:
there will be an exception that will be processed in the service and the server will continue working. null will be returned

sanioooook replied:
what should I use instead of SaveChanges?

horba commented on server/Entities/UserRepository.cs:
```diff
@@ -11,7 +12,7 @@ public class UserRepository
     public UserRepository(IRepositoryContext repositoryContext)
     {
-      this.RepositoryContext = repositoryContext;
+      RepositoryContext = repositoryContext;
```
_repositoryContext

sanioooook replied:
ok

horba commented on server/WebAPI/Controllers/ScheduledPlaceToEatController.cs:
```diff
@@ -0,0 +1,118 @@
+      {
+        foreach(var file in files)
+        {
+          if(!Consts.AllowedImageContentTypes.Contains(file.ContentType, StringComparer.OrdinalIgnoreCase) && !Consts.AllowedTextContentTypes.Contains(file.ContentType, StringComparer.OrdinalIgnoreCase))
```
might you need model validators?

sanioooook replied:
I could not make the server accept at least some file, so the model validator I don't think I can do it

horba replied:
might, you try?
https://stackoverflow.com/questions/46694745/asp-net-core-custom-model-validation

sanioooook replied:
ok

horba commented on server/WebAPI/Controllers/ScheduledPlaceToEatController.cs:
```diff
@@ -0,0 +1,118 @@
+      NamePlace = scheduledPlaceToEatDTO.NamePlace,
+      Notes = scheduledPlaceToEatDTO.Notes
+      });
+    }
```
use mapper

sanioooook replied:
ok

demosglok commented on front/src/constants/validation.js:
```diff
@@ -1,5 +1,8 @@
-const EMAIL_REGEX = '^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$';
+const EMAIL_REGEX = '^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$',
+      LINK_REGEX = '^((?:https?:)?(?:\\/{2})?)?((?:[\\w]{1,64})\\.(?:[\\w\\.]{2,64}))'
```
could you plz put comments with an example of email and link, passing this regex

sanioooook replied:
ok

demosglok commented on front/src/filters/google-place-types-filter.js:
```diff
@@ -0,0 +1,27 @@
+    if (!type) {
+      return type;
+    }
+    if (type === 'food') {
```
switch-case?

demosglok commented on front/src/locales/translations/enUS.json:
```diff
@@ -185,6 +185,41 @@
+    "delete": "Delete",
+    "successfullyDelete": "Successfully delete"
+  },
+  "eatingNew": {
```
how is "eatingNew" different from "eating"

sanioooook replied:
these are two different pages: one shows already planned places, the second creates new or edits existing places to eat

horba replied:
might, eating - is not ideal name - that\s why these question apeared

demosglok commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+export default {
+  namespaced: true,
+  actions: {
+    getEatingUser ({ commit }) {
```
do you need "commit" here?

sanioooook replied:
no, I fix it

demosglok commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+    getEatingUser ({ commit }) {
+      return api.get('/scheduledPlaceToEat');
+    },
+    getEatingUserByEatingId ({ commit }, payload) {
```
are you using "commit" in this function?

sanioooook replied:
getEatingUser - unused 
getEatingUserByEatingId - unused, but need, because if I delete '{ commit}, ', payload be = first object which has the property commit

demosglok commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+          return scheduledPlaceToEat;
+        }).catch(error => Promise.reject(error));
+    },
+    createNewEating ({ commit }, payload) {
```
we're not using "commit" inside this action, are we?

sanioooook replied:
we don't use it, but payload is passed only by the second parameter, and if the first parameter is { }, there will be an error

horba replied:
of course, because `{ }` means - what? empty instance of object, how can we write in SIGNATURE instance - you can't write instance of object - only assign name of objects which passed as paramater to function. when you write { commit } - it is DESTRUCTURING ASSIGMENT
please, think what it is signature of function, `{ }` and `{ commit }` - and when we shall use Destructuring assignment and what it means at all.

horba replied:
for unused params names you can use `_` as parameter name or as prefix paramater name, so it could be
`getEatingUserByEatingId (_, payload) {`
or
`getEatingUserByEatingId (_context, payload) {`
if it is alone you can use first option (like in this case)

demosglok commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+      return api.post('/ScheduledPlaceToEat/create/', payload);
+    },
+    updateEating ({ commit }, payload) {
+      return api.post('/scheduledPlaceToEat/update/', payload);
```
same

demosglok commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+    },
+    deleteEating ({ commit }, payload) {
+      if (payload.id) {
+        return api.post('/scheduledPlaceToEat/deleteById/' + payload.id);
```
same

demosglok commented on front/src/store/modules/eating.store.js:
```diff
@@ -0,0 +1,48 @@
+        return api.post('/scheduledPlaceToEat/delete/', payload);
+      }
+    },
+    uploadNewFilesEating ({ commit }, { eatingId, files }) {
```
same

demosglok commented on front/src/views/user/eating/eating-new/eating-new.view.html:
```diff
@@ -0,0 +1,108 @@
+        background-color="white"
+        dense
+        :label="$t('eatingNew.addFile')"
+        accept="image/*, .txt, .doc, .docx"
```
pdf?

sanioooook replied:
ok

demosglok commented on front/src/views/user/eating/eating-new/eating-new.view.html:
```diff
@@ -0,0 +1,108 @@
+    {{$t('eatingNew.close')}}
+      </v-btn>
+  </v-snackbar>
+</v-form>
```
no new line at the end of file

sanioooook replied:
ok

demosglok commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+                valid = false;
+              }
+            });
+            return valid ? true : 'File size should be less than 10 MB!';
```
i18n?

sanioooook replied:
ok

demosglok commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+        },
+        fileLength: v => {
+          if (v) {
+            return v.length <= 10 || 'Количество вложений не должно быть больше 10';
```
i18n?

sanioooook replied:
ok

demosglok commented on front/src/views/user/eating/eating-new/eating-new.view.js:
```diff
@@ -0,0 +1,171 @@
+      time: new Date().toTimeString().substr(0, 5),
+      notes: '',
+      link: '',
+      namePlace: '',
```
placeName?

sanioooook replied:
ok

demosglok commented on front/src/views/user/eating/eating.view.html:
```diff
@@ -0,0 +1,109 @@
+        <v-img v-if="place.googleDetails && place.googleDetails.photos" 
+          :aspect-ratio="16/9"
+          :lazy-src="defaultImage" :src="place.googleDetails.photos[0].getUrl(259)"
+          max-height="150" max-width="259" />
```
why 259? what is it?

horba replied:
and does it connected with server validation value?

demosglok commented on front/src/views/user/eating/eating.view.html:
```diff
@@ -0,0 +1,109 @@
+        <v-img v-else :aspect-ratio="16/9" 
+          :lazy-src="defaultImage"
+          src="../../../assets/default-image.jpg"
+          max-height="150" max-width="259" />
```
again 259

demosglok commented on front/src/views/user/eating/eating.view.html:
```diff
@@ -0,0 +1,109 @@
+  <v-snackbar v-model="snackbar" :color="color" multi-line :timeout="3000" top>
+    {{ text }}
+  </v-snackbar>
+</div>
```
newline

sanioooook replied:
ok

demosglok commented on front/src/views/user/eating/eating.view.js:
```diff
@@ -0,0 +1,127 @@
+      options: {
+        zoom: 10,
+        center: {
+          lat: 48.459286,
```
what a magic numbers???

demosglok commented on front/src/views/user/eating/eating.view.js:
```diff
@@ -0,0 +1,127 @@
+      this.$store.dispatch('eating/getEatingUser')
+        .then(response => {
+          this.scheduledPlacesList = response.data
+          /// scheduledPlacesList: [{serverInfo, googleDetails}]
```
remove comments if they are not usefull

demosglok commented on server/Entities/Interfaces/IAttachmentFileEatingRepository.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public void DeleteAttachmentFileEating(int id);
+  }
+}
```
new line

demosglok commented on server/Entities/Interfaces/IAttachmentFileEatingRepository.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public void UpdateAttachmentFileEating(AttachmentFileEating attachmentFileEating);
+
+    public void DeleteAttachmentFileEating(AttachmentFileEating attachmentFileEating);
```
do we need this method if we have latter one with id?

demosglok commented on server/Entities/Interfaces/IAttachmentFileEatingRepository.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public void CreateAttachmentFileEating(AttachmentFileEating attachmentFileEating);
+
+    public void UpdateAttachmentFileEating(AttachmentFileEating attachmentFileEating);
```
will this throw if attachmentFileEating will have empty id?

sanioooook replied:
the object is validated before the repository (in the service)

demosglok commented on server/Entities/Interfaces/IAttachmentFileEatingRepository.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public IEnumerable<AttachmentFileEating> GetAttachmentFileEatingByScheduledPlaceId(int ScheduledPlaceId);
+
+    public void CreateAttachmentFileEating(AttachmentFileEating attachmentFileEating);
```
will this throw if attachmentFileEating has nonempty id?

sanioooook replied:
verification takes place in the service, repository only working with the database

demosglok commented on server/Entities/Interfaces/IScheduledPlaceToEatRepository.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public void UptateScheduledPlaceToEat(ScheduledPlaceToEat scheduledPlaceToEat);
+
+    public void DeleteScheduledPlaceToEat(ScheduledPlaceToEat scheduledPlaceToEat);
```
?

demosglok commented on server/Entities/Models/AttachmentFileEating.cs:
```diff
@@ -0,0 +1,21 @@
+    [ForeignKey("ScheduledPlaceToEatId")]
+    public ScheduledPlaceToEat ScheduledPlaceToEat { get; set; }
+  }
+}
```
new line

demosglok commented on server/Entities/Models/ScheduledPlaceToEat.cs:
```diff
@@ -0,0 +1,40 @@
+    [DataType(DataType.Url, ErrorMessage = "It doesn't look like a link")]
+    public string Link { get; set; }
+
+    [StringLength(50, ErrorMessage = "{0} length must be up to {1}.")]
```
is it translated?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Assets/AssetsController.cs:
```diff
@@ -31,12 +37,23 @@ public async Task<IActionResult> UploadFile([FromForm] IFormFile file, [FromRout
+        case AssetType.UserAvatar:
+        {
+          if(file.Length > Consts.MaxImageFileSize)
+            return BadRequest("fileIsToBig");
```
what is "fileIsToBig" ? first, "fileIsTooBig", second, why is it a string and how is it translated into meaningfull message?

demosglok commented on server/WebAPI/Controllers/ScheduledPlaceToEatController.cs:
```diff
@@ -0,0 +1,118 @@
+
+          if(file.Length > Consts.MaxEatingFileSize)
+          {
+            return BadRequest("fileIsToBig");
```
fileIsTooBig

demosglok commented on server/WebAPI/Controllers/ScheduledPlaceToEatController.cs:
```diff
@@ -0,0 +1,118 @@
+    [HttpPost("Delete")]
+    public IActionResult Delete([FromBody] InputScheduledPlaceToEatForUpdateOrDeleteDTO ScheduledPlaceToEatDTO)
+    {
+      ScheduledPlaceToEatDTO.UserId = HttpContext.GetUserIdFromClaim();
```
why do you need userId here?

sanioooook replied:
to understand that the user deletes their own, and not someone else's plans

demosglok commented on server/WebAPI/DTO/ScheduledPlaceToEat/InputScheduledPlaceToEatForUpdateOrDeleteDTO.cs:
```diff
@@ -0,0 +1,36 @@
+
+namespace WebAPI.DTO.ScheduledPlaceToEat
+{
+  public class InputScheduledPlaceToEatForUpdateOrDeleteDTO
```
why do we need DTO for deleting, not just ID?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Interfaces/IScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,23 @@
+
+    public bool UpdateScheduledPlaceToEat(InputScheduledPlaceToEatForUpdateOrDeleteDTO scheduledPlaceToEatDTO);
+
+    public bool DeleteScheduledPlaceToEat(InputScheduledPlaceToEatForUpdateOrDeleteDTO scheduledPlaceToEatDTO);
```
do we need deletion by dto, not id?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/Assets/Consts.cs:
```diff
@@ -3,12 +3,23 @@
+
     public static string[] AllowedImageContentTypes = new[] { "image/png", "image/jpeg", "image/jpg",  "image/bmp" };
 
+    public static string[] AllowedTextContentTypes = new[] { ".txt", ".doc", ".docx" };
```
pdf?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/AttachmentFileEatingService.cs:
```diff
@@ -0,0 +1,63 @@
+      {
+        if(file.Length >= Consts.MaxEatingFileSize)
+        {
+          return;
```
exception? warning? some notification for user that file is too big?

demosglok commented on server/WebAPI/Services/AttachmentFileEatingService.cs:
```diff
@@ -0,0 +1,63 @@
+        foreach(var file in _attachmentFileEatingRepository.GetAttachmentFileEatingByScheduledPlaceId(eatingId))
+        {
+          _assetsService.DeleteFile(file.Path, AssetType.FileEating);
+          _attachmentFileEatingRepository.DeleteAttachmentFileEating(file);
```
isn't file.id enough here?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/AttachmentFileEatingService.cs:
```diff
@@ -0,0 +1,63 @@
+      if(_userService.GetUser(userId) != null && attachmentFileEating != null && attachmentFileEating.ScheduledPlaceToEat.UserId == userId)
+      {
+        _assetsService.DeleteFile(attachmentFileEating.Path, AssetType.FileEating);
+        _attachmentFileEatingRepository.DeleteAttachmentFileEating(attachmentFileEating);
```
isn't attachmentFileEating.id enough here?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+      }
+      catch(Exception)
+      {
+        return -1;
```
what does this -1 mean?

sanioooook replied:
-1 indicates that the creation was not successful and the controller will analyze this and return an error

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+        List<OutputScheduledPlaceToEatDTO> result = new List<OutputScheduledPlaceToEatDTO>();
+        foreach(var eating in eatings)
+        {
+          result.Add(ConvertScheduledPlaceToEatToOutputScheduletPlaceToEatDTO(eating));
```
linq+map instead of foreach and manual population of result?

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+      }
+      catch(Exception e)
+      {
+        Console.WriteLine(e);
```
any message for user why this didn't worked out?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+      catch(Exception e)
+      {
+        Console.WriteLine(e);
+        return false;
```
any explanation for caller?

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+      var attachmentFileEatings = _attachmentFileEatingRepository.GetAttachmentFileEatingByScheduledPlaceId(scheduledPlaceToEat.Id);
+      if(attachmentFileEatings != null)
+      {
+        foreach(var file in attachmentFileEatings)
```
map?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+        {
+          foreach(var error in results)
+          {
+            Console.WriteLine(error.ErrorMessage);
```
why do we need to do console.writeline here?

sanioooook replied:
ok

demosglok commented on server/WebAPI/Services/ScheduledPlaceToEatService.cs:
```diff
@@ -0,0 +1,254 @@
+        {
+          foreach(var error in results)
+          {
+            Console.WriteLine(error.ErrorMessage);
```
why do we need to do console.writeline here?

sanioooook replied:
ok

horba commented on front/src/views/user/eating/eating-new/eating-new.view.html:
```diff
@@ -0,0 +1,108 @@
+<v-form v-model="formValid">
```
isValid

sanioooook replied:
ok

horba commented on server/WebAPI/Controllers/ScheduledPlaceToEatController.cs:
```diff
@@ -0,0 +1,118 @@
+    public IActionResult Delete( int id)
+    {
+      var scheduledPlaceToEatDTO = _scheduledPlaceToEatService.GetEatingByUserId(HttpContext.GetUserIdFromClaim()).Where(r => r.Id.Equals(id)).FirstOrDefault();
+      return Delete(new InputScheduledPlaceToEatForUpdateOrDeleteDTO {
```
i really don't understand why you need for deleting and updating dto

sanioooook replied:
the id property is optional for creating a dto. for dto, the id property is required for updating or deleting. but since I now only use the id from Route for deletion, this model will not be used for deletion, only for update

horba commented on front/src/views/user/eating/eating.view.js:
```diff
@@ -0,0 +1,127 @@
+    trimString (value) {
+      return value.length > 10 ? value.substr(0, 10) + '...' : value;
+    }
+  }
```
why not in common?

sanioooook replied:
ok

sanioooook committed:
fix: eating name changed to scheduledPlaceToEat
file paths via @
add: names of some places from Google places API
for example food or bar
new validators for the file (s) on the server side
new client-side validators for links and files
removed an extra method for removing places to eat
now to delete a place to eat, you need to pass the id

sanioooook committed:
Merge branch 'dev' of https://github.com/horba/my-trip into feature/73

sanioooook committed:
fix: rename file from image
usage on controller validators files
use in store postFile for upload files
rename some variables
add: filter trim string

sanioooook committed:
Merge branch 'dev' of https://github.com/horba/my-trip into feature/73

sanioooook committed:
code cleaning

horba reviewed (commented):
better to refactor services at backend
